### PR TITLE
Refactor cluster First and Join seed node process common logic

### PR DESCRIPTION
**Pull request by helena**
Kind of tech debt to clean things up, just something I looked at some weekend:
 Refactor First and Join seed node processe common logic for dry and logic hardening reasons.

### CODE REVIEW

**Review by patriknw — COMMENTED** [review]
good refactoring, I compared the methods with the original and it's looking good

File: akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala [hunk 1]
```diff
@@ -1502,6 +1502,140 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
 
 }
 
+/**
+ * INTERNAL API.
+ */
+@InternalApi
+private[cluster] abstract class SeedNodeProcess(joinConfigCompatChecker: JoinConfigCompatChecker) extends Actor {
```
[the remark this answers is not in the document]
Comment (helena) [reply akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala]: Sounds good.

**Comment by helena** [comment]
Split out to new seed node file, running on a multijvm Jenkins job.

**Review by raboof — APPROVED** [review]
Neat!

**Comment by raboof** [comment]
easier to review when you look at the first commit only, the second commit just moves the changed code to a separate file.

needs a rebase since it gained a conflict

**Comment by helena** [comment]
Closing in favor of https://github.com/akka/akka/pull/28379.

**Pull request closed without merging**